src/transformers/modeling_utils.py

```diff
@@ -551,7 +551,7 @@ def _load_state_dict_into_meta_model(
 
         module_name = param_name
         # We convert floating dtypes to the `dtype` passed.
-        if dtype is not None and not str(param.dtype).startswith("torch.int"):
+        if dtype is not None and not str(param.dtype).startswith(("torch.int", "torch.uint", "torch.bool")):
```

Comment:
Would that work?

```suggestion
        if dtype is not None and param.dtype.is_floating_point:
```
https://pytorch.org/docs/stable/generated/torch.is_floating_point.html#torch.is_floating_point

Reply:
Great catch! Let me try and I'll let you know 💪

Reply:
The slow tests seems pass under `accelerate-0.12.0` (latest release) ! 🎉 
Just added more comments above the if condition